intel_extension_for_transformers/llm/runtime/graph/models/gptneox/gptneox.cpp

```diff
@@ -135,41 +139,60 @@ static bool gptneox_model_eval_internal(model_context& lctx, const model_token*
 
         cur = ne_add(ctx0, ne_repeat(ctx0, model.layers[il].attn[1], cur), cur);
       }
-
-      struct ne_tensor* Qcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, n_embd / n_head, n_head, N, cur->nb[1] / n_head,
-                                                        cur->nb[1], 0 * sizeof(float) * n_embd / n_head));
-      struct ne_tensor* Kcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, n_embd / n_head, n_head, N, cur->nb[1] / n_head,
-                                                        cur->nb[1], 1 * sizeof(float) * n_embd / n_head));
-      struct ne_tensor* Vcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, n_embd / n_head, n_head, N, cur->nb[1] / n_head,
-                                                        cur->nb[1], 2 * sizeof(float) * n_embd / n_head));
+      struct ne_tensor* Qcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, head_dim, n_head, N * batch_size,
+                                                        cur->nb[1] / n_head, cur->nb[1], 0 * sizeof(float) * head_dim));
+      struct ne_tensor* Kcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, head_dim, n_head, N * batch_size,
+                                                        cur->nb[1] / n_head, cur->nb[1], 1 * sizeof(float) * head_dim));
+      struct ne_tensor* Vcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, head_dim, n_head, N * batch_size,
+                                                        cur->nb[1] / n_head, cur->nb[1], 2 * sizeof(float) * head_dim));
 
       // using mode = 2 for GPT-NeoX mode
-      Qcur = ne_rope_inplace(ctx0, Qcur, n_past, n_rot, 2, 0);
-      Kcur = ne_rope_inplace(ctx0, Kcur, n_past, n_rot, 2, 0);
+      Qcur = ne_rope_inplace(ctx0, ne_reshape_4d(ctx0, Qcur, head_dim, n_head, N, batch_size), n_past, n_rot, 2, 0);
+      Kcur = ne_rope_inplace(ctx0, ne_reshape_4d(ctx0, Kcur, head_dim, n_head, N, batch_size), n_past, n_rot, 2, 0);
```

Comment:
Yes, you are right. I was not sure how to make the `nb` right in L142-L147 since `gptneo-x` had a different `QKV` layout like the others. I will consider to optimize it if needed later. Thanks